src/app/core/auth/state/auth-state.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Action, State, StateContext } from '@ngxs/store'
+import { Injectable } from '@angular/core'
+import { AuthService } from 'src/app/core/auth/services/auth.service'
+import { Observable, tap } from 'rxjs'
+import { Tokens } from 'src/app/core/auth/types/tokens'
```

Comment:
Why import things from the root? You're already in the `core` folder. Use relative imports if your dependencies are in the same folder/module. Or even better – learn about [Typescript path aliasing](https://webreaper.dev/posts/tsconfig-paths-setup/).